modules/javafx.graphics/src/main/java/com/sun/marlin/OffHeapArray.java

```diff
@@ -26,82 +26,164 @@
 package com.sun.marlin;
 
 import static com.sun.marlin.MarlinConst.LOG_UNSAFE_MALLOC;
-import java.lang.reflect.Field;
-import sun.misc.Unsafe;
+import java.lang.foreign.Arena;
+import java.lang.foreign.MemorySegment;
+import java.lang.foreign.ValueLayout;
+import java.nio.ByteOrder;
+
+// KCR: BEGIN DEBUG
+import java.util.concurrent.atomic.AtomicInteger;
+// KCR: END DEBUG
 
-/**
- *
- */
-// FIXME: We must replace the terminally deprecated sun.misc.Unsafe
-// memory access methods; see JDK-8334137
-@SuppressWarnings("removal")
 final class OffHeapArray  {
 
-    // unsafe reference
-    static final Unsafe UNSAFE;
+    private Arena arena;
+
     // size of int / float
     static final int SIZE_INT;
+    // FFM stuff
+    private static final ValueLayout.OfByte BYTE_LAYOUT = ValueLayout.JAVA_BYTE;
+    private static final ValueLayout.OfInt INT_LAYOUT = ValueLayout.JAVA_INT.withOrder(ByteOrder.BIG_ENDIAN);
```

Comment:
@minborg Is this the best layout to use? I chose it thinking that, since we don't need to access this memory from native code, it might perform better to use Java's big endian byte order. Is this correct?